bin/offline-dbg

```diff
@@ -133,6 +142,143 @@ EOF
     echo "**************************************************************"
 }
 
+get_highest_ofp_version() {
+    # If no protocol listed, default to OpenFlow14 (enabled by default on ovs 2.8 on)
+    hex_ofp_version=$(cat ${CMD_DIR}/ovs-ofctl_-t_5_--version | grep "OpenFlow versions" | grep -oP [0-9]*x[0-9]* | tail -1)
+    version=${open_flow_protocols[$hex_ofp_version]}
+    if [ -z "$version" ]; then
+        echo "OpenFlow14"
+    else
+        echo $version
+    fi
+}
+
+do_collect-sos() {
+    OPTIND=1
+    while getopts "h" opt; do
+        case ${opt} in
+            h)
+                usage
+                exit 0
+                ;;
+        esac
+    done
+    shift $(((OPTIND -1)))
+
+    if [ $# -lt 1 ]; then
+        usage
+        exit 1
+    fi
+    local SOS=$1
+
+    Dir_name=`tar -tf $SOS | head -1 | cut -f1 -d"/"`
+    sos_files="$Dir_name/sos_commands/rpm/package-data 
+    $Dir_name/sos_commands/openvswitch
+    $Dir_name/var/lib/openvswitch
+    $Dir_name/etc/openvswitch
+    $Dir_name/usr/local/etc/openvswitch"
+
+    if test -f "$SOS"; then
+        mkdir ${WORKDIR}/sos_report
+        echo "Extracting sos report from '$SOS'..."
+        tar -xvf $SOS -C ${WORKDIR}/sos_report $sos_files --strip-components 1 --ignore-failed-read &>/dev/null || echo "Error occured while extracting specified \$sos_files from $SOS"
+        CMD_DIR=${WORKDIR}/sos_report/sos_commands/openvswitch/
```

Comment:
should we not fail in case of error here?

Reply:
So my thinking was that I want to grab all of the potential db locations, however, one potential location for one sos report may not exist in another, which would cause this to fail. I could just grab less specific directories, but then it would require more redundant extraction. I'm sure there is a better way to do this that I am just missing?

Reply:
In that case there will always an "Error .." message printed.
We could loop over all $sos_files and extract them one by one. That way if none of them extract successfully we can error out

Reply:
Right. I think this new commit should solve it. Fail if any critical files (ovs commands) are missing. Warn user is db is not found in specified location so they can still extract everything else and add the db manually with collect-db or by adding the location to the script manually.

Reply:
Please squash both commits as the latter only modifies code introduced by the former, fixes problems introduced by the former and has no commit comment